scalding-beam/src/test/scala/com/twitter/scalding/beam_backend/BeamBackendTests.scala

```diff
@@ -406,6 +449,7 @@ class BeamBackendTests extends FunSuite with BeforeAndAfter {
       .mapValues { case (left, right) => left + right }
       .filter(_._1 % 5 == 0)
       .toIterableExecution
+      .flatMap(iter => Execution.from(iter.toArray))
```

Comment:
After adding logic of deletion, this started to fail with files not found. This is because iterator is reading the output files as it is. For what I know is that we don't use `toIterableExecution` as a final step, we always map the iterator to do another computation.

Reply:
oh... okay. I think the problem is that we are closing over a temporary file. So, a final `toIterableExecution` can't be lazy, it has to go ahead and materialize if we are going to clean the iterator outputs.

Do you think that makes sense?